src/components/Upload/Upload.stories.tsx

```diff
@@ -507,6 +513,77 @@ const Image_Editor_Story: ComponentStory<typeof Upload> = () => {
   );
 };
 
+const Basic_Deferred_API_Story: ComponentStory<typeof Upload> = () => {
+  const [data, setData] = useState<Record<string, unknown>>({});
+  const [thumbUrl, setThumbUrl] = useState<string>('');
+
+  const [fileList, setFileList] = useState<UploadFile[]>([]);
+
+  const onChange: UploadProps['onChange'] = async (info: any) => {
+    await generateThumbAsync(info.file).then(() => {
+      setData({
+        ...info.file,
+        name: info.file.name,
+        percent: 100,
+        status: 'done',
+        thumbUrl: info.file.preview,
+      });
+    });
+  };
+
+  const getBase64 = (file: OcFile): Promise<string> =>
+    new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.readAsDataURL(file);
+      reader.onload = () => resolve(reader.result as string);
+      reader.onerror = (error) => reject(error);
+    });
+
+  const generateThumbAsync = async (file: UploadFile): Promise<void> => {
+    if (!file.url && !file.preview) {
+      file.preview = await getBase64(file.originFileObj as OcFile);
+    }
+    setThumbUrl(file.url || (file.preview as string));
+  };
+
+  const props: UploadProps = {
+    customRequest() {
+      return true;
+    },
+    data: data,
```

Comment:
I see, so we could have provided data prop to upload component for the images to show